### PR TITLE
Tighten types that were reported by phpstan.

### PR DESCRIPTION
**Pull request by biozshock**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ x ]
| Deprecations?                          | [ x ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ x ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:
Tightened types and fixed phpstan errors. 

There are still some PHPStan errors revealed by the type tightening or method overrides. Best solution as for me is to generate new baseline.

#### Steps to test this PR:
As there are lots of places changed, please test as much as you can.

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Check the installation works.

### CODE REVIEW

File: app/bundles/CampaignBundle/Command/TriggerCampaignCommand.php [hunk 1]
```diff
@@ -167,6 +168,34 @@ protected function execute(InputInterface $input, OutputInterface $output): int
         $threadId      = $input->getOption('thread-id');
         $maxThreads    = $input->getOption('max-threads');
 
+        if (is_numeric($id)) {
```
Comment (kuzmany) [inline app/bundles/CampaignBundle/Command/TriggerCampaignCommand.php:179]: Shouldn't !is_numberic condition?
Based on code make more sense
Other places as well

**Review by kuzmany — CHANGES_REQUESTED** [review]

File: app/bundles/CampaignBundle/Command/TriggerCampaignCommand.php (hunk 1 above)
Comment (biozshock) [reply app/bundles/CampaignBundle/Command/TriggerCampaignCommand.php:179]: @kuzmany The reason for this check is that only numeric values will be casted to `int` properly. Non numeric values should raise a type error, or have an error message in `else`.

Comment (kuzmany) [reply app/bundles/CampaignBundle/Command/TriggerCampaignCommand.php:179]: I see it as a bit unnecessary, but ok. Directly integer type  declaration would be cleaner for me.

Comment (kuzmany) [reply app/bundles/CampaignBundle/Command/TriggerCampaignCommand.php:179]: @escopecz your opinion?

File: app/bundles/CoreBundle/Factory/MauticFactory.php [hunk 2]
```diff
@@ -341,14 +347,20 @@ public function getLogger($system = false)
...
     {
-        return match ($helper) {
-            'template.assets'     => $this->container->get('twig.helper.assets'),
-            'template.slots'      => $this->container->get('twig.helper.slots'),
-            'template.form'       => $this->container->get('twig.helper.form'),
-            'template.translator' => $this->container->get('twig.helper.translator'),
-            'template.router'     => $this->container->get('twig.helper.router'),
-            default               => $this->container->get('mautic.helper.'.$helper),
-        };
+        switch ($helper) {
```
Comment (escopecz) [inline app/bundles/CoreBundle/Factory/MauticFactory.php:357]: Why was this changed back from match to switch and the options are different?

File: app/bundles/CampaignBundle/Command/TriggerCampaignCommand.php (hunk 1 above)
Comment (escopecz) [reply app/bundles/CampaignBundle/Command/TriggerCampaignCommand.php:179]: It depends wether you intend to use the cast to int as a user value sanitization. How will this work with

`bin/console mautic:campaigns:trigger --campaign-id=unicorn`

Is the code fine with working with strings? Is retyping the string to an int valid? Quick test:
```
php > var_dump((int) 'unicorn');
int(0)
php > var_dump((int) '100unicorn');
int(100)
```
Ideally, it should guide the user and reject invalid values. But that would deserve its own PR.

I understand what @biozshock meant to accomplish here. This way it won't convert `null` values to `0`. But I understand what @kuzmany means too. There are other values than `int` and `null`.

File: app/bundles/CoreBundle/Factory/MauticFactory.php (hunk 2 above)
Comment (biozshock) [reply app/bundles/CoreBundle/Factory/MauticFactory.php:357]: @escopecz The goal was to replace `container::get` (is deprecated), and to tighten services that are used in the `getHelper`. If i didn't miss any, then those are only ones that are used with the method.
I couldn't replace this with match, because on "default" we should warn users, at least in test case, otherwise rector should replace this with match.

**Review by escopecz — COMMENTED** [review]
This PR is improving the code, but the plan was to release M5.0.0 stable next week. If we'd merge this then we'd need to release new release candidate version as it contains BC breaks. What would you suggest to do here?

**Comment by kuzmany** [comment]
@escopecz As I see, it's baked on the 5.1 branch, so I assume we can go to 5.1.

**Comment by escopecz** [comment]
Breaking changes can go to 6.x from now on. If someone wants to maintain that branch for a year or so, we can create it now.

**Comment by escopecz** [comment]
I changed the target branch from 5.1 to 5.x as we are getting back to our classic branching strategy where 5.x will be for features and 5.0 for bug fixes.

There is a conflict. Please resolve it.

File: app/bundles/CampaignBundle/Command/TriggerCampaignCommand.php [hunk 3]
```diff
@@ -167,6 +168,34 @@ protected function execute(InputInterface $input, OutputInterface $output): int
...
+            $id = (int) $id;
+        }
+
+        if (is_numeric($maxThreads)) {
+            $maxThreads = (int) $maxThreads;
+        }
```
Comment (escopecz) [inline app/bundles/CampaignBundle/Command/TriggerCampaignCommand.php:185]: As we need either `int` or `null` in the`ContactLimiter`, how about to do this?
```suggestion
        $maxThreads = is_numeric($maxThreads) ? (int) $maxThreads : null;
```

File: app/bundles/CampaignBundle/Command/TriggerCampaignCommand.php (hunk 1 above)
Comment (escopecz) [reply app/bundles/CampaignBundle/Command/TriggerCampaignCommand.php:179]: I'm suggesting this: https://github.com/mautic/mautic/pull/13110/files#r1453534132

File: app/bundles/CampaignBundle/Command/TriggerCampaignCommand.php (hunk 3 above)
Comment (biozshock) [reply app/bundles/CampaignBundle/Command/TriggerCampaignCommand.php:185]: I think the changes you propose should be a separate PR.
Every command should control the type of the option/argument with the appropriate message. For example if you set the `max-threads` option to _aaaa_ then the application should exit with an error code and an appropriate message.
Otherwise _some_ users could be confused why the application does not work as they would expect that `contact-id`, `min-contact-id`, etc. instead of giving an error is producing unexpected results converting the option is either a numeric or null.

Otherwise this PR should contain messages for affected commands, or extract command-related type changes into a separate PR and do it there.

Comment (escopecz) [reply app/bundles/CampaignBundle/Command/TriggerCampaignCommand.php:185]: You are right. If they send `aaa` to an int param then they get an exception that somewhere in the code we expect an int and got an string. Better than silently resetting the param.

**Comment by escopecz** [comment]
@biozshock can you either regenerate the baseline or add comment ignores to the code with the explanations why is the line ignored or add the ignores to phpstan configs so we could merge this? I'm fine with either of those.

**Comment by biozshock** [comment]
@escopecz What is better or common in Mautic? I can do both of the solutions to get phpstan errors out of the way.

File: app/bundles/PluginBundle/Tests/ConfigFormTest.php [hunk 4]
```diff
@@ -44,7 +45,7 @@ public function testOauth(): void
         $parameters = ['a' => 'testa', 'b' => 'testb'];
         $method     = 'GET';
         $authType   = 'oauth2';
-        $expected   = [
+        $expectedT  = [
```
Comment (escopecz) [inline app/bundles/PluginBundle/Tests/ConfigFormTest.php]: Is this a typo?

File: plugins/MauticSocialBundle/Command/MauticSocialMonitoringCommand.php [hunk 5]
```diff
@@ -47,8 +47,14 @@ protected function execute(InputInterface $input, OutputInterface $output): int
             return \Symfony\Component\Console\Command\Command::SUCCESS;
         }
 
+        if (!is_numeric($batchSize)) {
+            $output->writeln('batch-size is not number.');
+
+            return 1;
```
Comment (escopecz) [inline plugins/MauticSocialBundle/Command/MauticSocialMonitoringCommand.php]:
```suggestion
            return \Symfony\Component\Console\Command\Command::FAILURE;
```

File: app/bundles/PluginBundle/Tests/ConfigFormTest.php (hunk 4 above)
Comment (biozshock) [reply app/bundles/PluginBundle/Tests/ConfigFormTest.php]: Was a leftover. Removed.

File: plugins/MauticSocialBundle/Command/MauticSocialMonitoringCommand.php (hunk 5 above)
Comment (biozshock) [reply plugins/MauticSocialBundle/Command/MauticSocialMonitoringCommand.php]: Though i used `self::FAILURE`, which is shorter. :)

File: app/bundles/CoreBundle/Test/MauticMysqlTestCase.php [hunk 6]
```diff
@@ -383,8 +382,7 @@ private function markDatabasePrepared(): void
 
     private function clearCache(): void
     {
-        $cacheProvider = self::$container->get('mautic.cache.provider');
-        \assert($cacheProvider instanceof CacheItemPoolInterface);
```
Comment (escopecz) [inline app/bundles/CoreBundle/Test/MauticMysqlTestCase.php]: Can we keep the assert so the static analytics tools know what type of variable this is?

File: app/bundles/CoreBundle/Tests/Functional/Doctrine/Paginator/SimplePaginatorTest.php [hunk 7]
```diff
@@ -24,8 +24,7 @@ protected function setUp(): void
     {
         parent::setUp();
 
-        $debugDataHolder = self::$container->get('doctrine.debug_data_holder');
-        \assert($debugDataHolder instanceof DebugDataHolder);
```
Comment (escopecz) [inline app/bundles/CoreBundle/Tests/Functional/Doctrine/Paginator/SimplePaginatorTest.php]: Can we keep the assert so the static analytics tools know what type of variable this is?

File: app/bundles/CoreBundle/Tests/Functional/Helper/LanguageHelperTest.php [hunk 8]
```diff
@@ -12,12 +12,11 @@ final class LanguageHelperTest extends MauticMysqlTestCase
 {
     public function testGettingLanguageFiles(): void
     {
-        $languageHelper = self::$container->get(LanguageHelper::class);
-        \assert($languageHelper instanceof LanguageHelper);
```
Comment (escopecz) [inline app/bundles/CoreBundle/Tests/Functional/Helper/LanguageHelperTest.php]: Can we keep the assert so the static analytics tools know what type of variable this is?

File: app/bundles/CoreBundle/Tests/Unit/Twig/Extension/AssetExtensionTest.php [hunk 9]
```diff
@@ -12,8 +12,7 @@ class AssetExtensionTest extends AbstractMauticTestCase
 {
     public function testGetCountryFlag(): void
     {
-        $assetExtension = self::$container->get(AssetExtension::class);
-        \assert($assetExtension instanceof AssetExtension);
```
Comment (escopecz) [inline app/bundles/CoreBundle/Tests/Unit/Twig/Extension/AssetExtensionTest.php]: Can we keep the assert so the static analytics tools know what type of variable this is?

File: app/bundles/CoreBundle/Tests/Unit/Twig/Extension/MenuExtensionTest.php [hunk 10]
```diff
@@ -13,8 +13,7 @@ class MenuExtensionTest extends AbstractMauticTestCase
 {
     public function testParseMenuAttributes(): void
     {
-        $menuExtension = self::$container->get(MenuExtension::class);
-        \assert($menuExtension instanceof MenuExtension);
```
Comment (escopecz) [inline app/bundles/CoreBundle/Tests/Unit/Twig/Extension/MenuExtensionTest.php]: Can we keep the assert so the static analytics tools know what type of variable this is?

File: app/bundles/CoreBundle/Tests/Unit/Twig/Extension/MenuExtensionTest.php [hunk 11]
```diff
@@ -29,8 +28,7 @@ public function testParseMenuAttributes(): void
 
     public function testBuildMenuClasses(): void
     {
-        $menuExtension = self::$container->get(MenuExtension::class);
-        \assert($menuExtension instanceof MenuExtension);
```
Comment (escopecz) [inline app/bundles/CoreBundle/Tests/Unit/Twig/Extension/MenuExtensionTest.php]: Can we keep the assert so the static analytics tools know what type of variable this is?

File: app/bundles/EmailBundle/MonitoredEmail/Processor/Bounce/BouncedEmail.php [hunk 12]
```diff
@@ -114,10 +111,7 @@ public function setType($bounceType)
         return $this;
     }
 
-    /**
-     * @return int
-     */
-    public function isFinal()
+    public function isFinal(): bool
```
Comment (escopecz) [inline app/bundles/EmailBundle/MonitoredEmail/Processor/Bounce/BouncedEmail.php:114]: Please fix the return type in the comment. Adding it like this is considered a BC break as it changes the method signature.

File: app/bundles/EmailBundle/Tests/Model/EmailModelFunctionalTest.php [hunk 13]
```diff
@@ -49,8 +47,7 @@ private function generateContacts(int $howMany): array
             $contacts[] = $contact;
         }
 
-        $contactModel = self::$container->get('mautic.lead.model.lead');
-        \assert($contactModel instanceof LeadModel);
```
Comment (escopecz) [inline app/bundles/EmailBundle/Tests/Model/EmailModelFunctionalTest.php]: Can we keep the assert so the static analytics tools know what type of variable this is?

File: app/bundles/EmailBundle/Tests/Model/EmailModelFunctionalTest.php [hunk 14]
```diff
@@ -26,8 +25,7 @@ public function testSendEmailToListsInThreads(): void
         $this->addContactsToSegment($contacts, $segment);
         $email = $this->createEmail($segment);
 
-        $emailModel                                             =  self::$container->get('mautic.email.model.email');
-        \assert($emailModel instanceof EmailModel);
```
Comment (escopecz) [inline app/bundles/EmailBundle/Tests/Model/EmailModelFunctionalTest.php:38]: Can we keep the assert so the static analytics tools know what type of variable this is?

File: app/bundles/LeadBundle/Entity/Import.php [hunk 15]
```diff
@@ -560,11 +560,11 @@ public function getSatusLabelClass(): string
     }
 
     /**
-     * @param int $dateStarted
+     * @param \DateTimeInterface $dateStarted
      *
      * @return Import
      */
-    public function setDateStarted(\DateTime $dateStarted)
+    public function setDateStarted($dateStarted)
```
Comment (escopecz) [inline app/bundles/LeadBundle/Entity/Import.php]: This is also changing a public method signature

File: app/bundles/LeadBundle/Entity/Import.php [hunk 16]
```diff
@@ -681,10 +681,8 @@ public function getRunTimeSeconds()
 
     /**
      * Counts speed in items per second.
-     *
-     * @return float
      */
-    public function getSpeed()
+    public function getSpeed(): float
```
Comment (escopecz) [inline app/bundles/LeadBundle/Entity/Import.php:685]: Also here

File: app/bundles/LeadBundle/Tests/Deduplicate/ContactMergerFunctionalTest.php [hunk 17]
```diff
@@ -14,11 +14,11 @@ final class ContactMergerFunctionalTest extends MauticMysqlTestCase
 {
     public function testMergedContactFound(): void
     {
-        $model = self::$container->get('mautic.lead.model.lead');
-        \assert($model instanceof LeadModel);
+        /** @var LeadModel $model */
+        $model = self::getContainer()->get('mautic.lead.model.lead');
```
Comment (escopecz) [inline app/bundles/LeadBundle/Tests/Deduplicate/ContactMergerFunctionalTest.php]: The assert() function is preferred over a comment typehint as it will actually fail on test and dev environments yet has no affect on prod environments.

Other occurrences in this file.

File: app/bundles/LeadBundle/Tests/Entity/LeadFieldRepositoryFunctionalTest.php [hunk 18]
```diff
@@ -154,13 +154,13 @@ public function testCompareValueInOperatorWithMultiselectField(): void
         );
 
         $fieldModel = self::getContainer()->get(FieldModel::class);
-        \assert($fieldModel instanceof FieldModel);
+        \assert($fieldModel instanceof FieldModel); // @phpstan-ignore-line
```
Comment (escopecz) [inline app/bundles/LeadBundle/Tests/Entity/LeadFieldRepositoryFunctionalTest.php]: Why is this causing an issue for PHPSTAN? Can you add the reason for it after the ignore so it's clear? There is one more such occurrence in this file

**Review by escopecz — CHANGES_REQUESTED** [review]
Great changes here! I found a few more issues. Please take a look if it makes sense.

File: app/bundles/EmailBundle/MonitoredEmail/Processor/Bounce/BouncedEmail.php (hunk 12 above)
Comment (biozshock) [reply app/bundles/EmailBundle/MonitoredEmail/Processor/Bounce/BouncedEmail.php:114]: Is changed by the rector.

File: app/bundles/LeadBundle/Entity/Import.php (hunk 16 above)
Comment (biozshock) [reply app/bundles/LeadBundle/Entity/Import.php:685]: Changed by rector

File: app/bundles/EmailBundle/Entity/EmailRepository.php [hunk 19]
```diff
@@ -49,7 +49,7 @@ public function getDoNotEmailList($leadIds = []): array
      *
      * @param string $email
      *
-     * @return bool
+     * @return false|array{id: numeric-string, unsubscribed: bool, bounced: bool, manual: bool, comments: string}
```
Comment (mallezie) [inline app/bundles/EmailBundle/Entity/EmailRepository.php:61]: Do we really want this kind of return types. I would refactor to make sure we only return a boolean value. 
But for here i would keep this out of scope (this is already huge).

File: app/bundles/EmailBundle/Tests/Entity/PendingQueryFunctionalTest.php [hunk 20]
```diff
@@ -58,8 +57,7 @@ private function generateContacts(int $howMany): array
             $contacts[] = $contact;
         }
 
-        $contactModel = self::$container->get('mautic.lead.model.lead');
-        \assert($contactModel instanceof LeadModel);
+        $contactModel = self::getContainer()->get('mautic.lead.model.lead');
         $contactModel->saveEntities($contacts);
```
Comment (mallezie) [inline app/bundles/EmailBundle/Tests/Entity/PendingQueryFunctionalTest.php]: We should keep the assert here as well.

File: app/bundles/EmailBundle/Tests/Model/EmailModelFunctionalTest.php [hunk 21]
```diff
@@ -157,7 +155,7 @@ public function testNotOverwriteChildrenTranslationEmailAfterSaveParent(): void
        $this->em->detach($childrenEmail);
 
        /** @var EmailModel $emailModel */
-       $emailModel = self::$container->get('mautic.email.model.email');
+       $emailModel = self::getContainer()->get('mautic.email.model.email');
```
Comment (mallezie) [inline app/bundles/EmailBundle/Tests/Model/EmailModelFunctionalTest.php]: Not really in scope, but to be consistent, could we add an assert here.

File: app/bundles/LeadBundle/Entity/Import.php [hunk 22]
```diff
@@ -499,7 +499,7 @@ public function getProgressPercentage(): float|int
             return round(($processed / $total) * 100, 2);
         }
 
-        return 0;
+        return 0.0;
```
Comment (mallezie) [inline app/bundles/LeadBundle/Entity/Import.php:504]: Should this function now only return float instead of float|int. Not sure this is BC break, but otherwise we could add a return comment, so at least phpstan knows it returns floats, and can warn for something else.

File: app/bundles/LeadBundle/Tests/EventListener/ReportNormalizeSubscriberTest.php [hunk 23]
```diff
@@ -22,8 +22,8 @@ class ReportNormalizeSubscriberTest extends MauticMysqlTestCase
      */
     public function testOnReportDisplay(string $value, string $type, array $properties, string $expected): void
     {
-        $fieldModel = self::$container->get('mautic.lead.model.field');
-        \assert($fieldModel instanceof FieldModel);
+        $fieldModel = self::getContainer()->get('mautic.lead.model.field');
+        \assert($fieldModel instanceof FieldModel); // @phpstan-ignore-line
```
Comment (mallezie) [inline app/bundles/LeadBundle/Tests/EventListener/ReportNormalizeSubscriberTest.php]: Why do we have a phpstan ignore here. If this is not directly fixable, i would add it to the baseline. Also not sure what phpstan should not like here.

File: app/bundles/LeadBundle/Tests/Validator/Constraints/EmailAddressValidatorTest.php [hunk 24]
```diff
@@ -23,15 +22,14 @@ public function testValidate(?string $value, int $expectedViolationCount): void
         $emailAddressValidator = self::$container->get('mautic.validator.emailaddress');
         \assert($emailAddressValidator instanceof EmailAddressValidator);
 
-        $translator = self::$container->get('translator');
-        \assert($translator instanceof TranslatorInterface);
+        $translator = self::getContainer()->get('translator');
```
Comment (mallezie) [inline app/bundles/LeadBundle/Tests/Validator/Constraints/EmailAddressValidatorTest.php]: I think we should keep the assert here as well.

File: app/bundles/PageBundle/Controller/PublicController.php [hunk 25]
```diff
@@ -147,7 +148,7 @@ public function indexAction(
                         if (!empty($variantCookie)) {
                             if (isset($variants[$variantCookie])) {
                                 // if not the parent, show the specific variant already displayed to the visitor
-                                if ($variantCookie !== $entity->getId()) {
+                                if ((int) $variantCookie !== $entity->getId()) {
```
Comment (mallezie) [inline app/bundles/PageBundle/Controller/PublicController.php]: Why do we need this casting?

**Review by mallezie — CHANGES_REQUESTED** [review]
Left some comments when going through the code. Nothing major i think. 

In general this looks good, although the PR is huge, and very hard to review because of the size. 
There are still some test failures, i would propose to revert changes causing that, instead of further fixing them, since this is already huge.

File: app/bundles/EmailBundle/MonitoredEmail/Processor/Bounce/BouncedEmail.php (hunk 12 above)
Comment (escopecz) [reply app/bundles/EmailBundle/MonitoredEmail/Processor/Bounce/BouncedEmail.php:114]: I got the same problem on my PR. I solved it like this: https://github.com/mautic/mautic/pull/11251/commits/ef9a1896698c828f58d11d5422d01ca3957659c9

For M6 we'll remove that exclusion and let it do its work. I still don't want to disable that rule completely as it will help with all new methods having return types.

Comment (biozshock) [reply app/bundles/EmailBundle/MonitoredEmail/Processor/Bounce/BouncedEmail.php:114]: @escopecz I'd say this is a very bad idea. Because every new commit with new methods will not be checked and this will some devs will write already legacy code.

Comment (escopecz) [reply app/bundles/EmailBundle/MonitoredEmail/Processor/Bounce/BouncedEmail.php:114]: That's why I didn't disable the rule for the whole codebase but only for the 2 files that needed it.

@mollux what's your take on this? It's lesser evil to break the BC by allowing Rector to add the type or disabling the rule on those files to avoid the BC for the lifespan of M5?

File: app/bundles/LeadBundle/Entity/Import.php (hunk 22 above)
Comment (biozshock) [reply app/bundles/LeadBundle/Entity/Import.php:504]: @mallezie If i add the `float` as the only return type, then _rector_ changes this to the proper return type, and that's a BC break.

File: app/bundles/LeadBundle/Tests/EventListener/ReportNormalizeSubscriberTest.php (hunk 23 above)
Comment (biozshock) [reply app/bundles/LeadBundle/Tests/EventListener/ReportNormalizeSubscriberTest.php]: The @phpstan-ignore-line was there, because phpstan complained that the `assert` is useless. It's actually is :)

File: app/bundles/PageBundle/Controller/PublicController.php (hunk 25 above)
Comment (biozshock) [reply app/bundles/PageBundle/Controller/PublicController.php]: Yes, because the return type of the `$request->cookies->get()` is `bool|float|int|string|InputBag`. Want to make sure we are safe, when the cookie has a numeric string and strict equals will work as we expect. 
Best would be to have a check before this if with something like `if (!is_int($variantCookie) || !is_string($variantCookie) || !is_numeric($variantCookie)) { throw \RuntimeException }`. WDYT?

File: app/bundles/EmailBundle/Entity/EmailRepository.php (hunk 19 above)
Comment (biozshock) [reply app/bundles/EmailBundle/Entity/EmailRepository.php:61]: @escopecz @mollux WDYT?

Comment (escopecz) [reply app/bundles/EmailBundle/Entity/EmailRepository.php:61]: I'd say leave it as is. Still an improvement. Refactoring should be in a separate ticket.

File: app/bundles/EmailBundle/MonitoredEmail/Processor/Bounce/BouncedEmail.php (hunk 12 above)
Comment (biozshock) [reply app/bundles/EmailBundle/MonitoredEmail/Processor/Bounce/BouncedEmail.php:114]: @escopecz this is the only change that's left to change or keep. WDYT?

Comment (escopecz) [reply app/bundles/EmailBundle/MonitoredEmail/Processor/Bounce/BouncedEmail.php:114]: The chance that some plugin is extending such service is pretty small. I'd say let's close an eye when Rector is forcing us to add these obvious types.

File: app/bundles/PageBundle/Controller/PublicController.php (hunk 25 above)
Comment (biozshock) [reply app/bundles/PageBundle/Controller/PublicController.php]: Closing this conversation, because `$variantCookie` here can be `string|int` only, because array only accepts those AFAIK. And `$entity->getId()` will produce the `int` type.

Comment (escopecz) [reply app/bundles/PageBundle/Controller/PublicController.php]: I've seen many times that `$entity->getId()` returns `string`. And it should have if the ID is type of `BIGINT unsigned` as that can return a number greater than max PHP `int` number.

Comment (biozshock) [reply app/bundles/PageBundle/Controller/PublicController.php]: @escopecz Yes, sure there are cases where this line does not make sense, but in this case we have `\Mautic\PageBundle\Entity\Page::getId`, which returns an int (according to PHPDoc), also the `\Mautic\CoreBundle\Doctrine\Mapping\ClassMetadataBuilder::addId` adds an unsigned int and not a bigint.
Also, i'm not against changing this to `(string) $variantCookie !== (string) $entity->getId()`

Comment (escopecz) [reply app/bundles/PageBundle/Controller/PublicController.php]: Do not trust PHPDocs. They were copied from one place to another without any thought if they are true or not. I think that retyping both to the same type would make sense. I'm not sure if it matters if it is to int or string in this case.

Comment (biozshock) [reply app/bundles/PageBundle/Controller/PublicController.php]: Made it a string cast, to be very safe.

File: app/bundles/CampaignBundle/Tests/Controller/CampaignControllerFunctionalTest.php [hunk 26]
```diff
@@ -46,7 +46,11 @@ protected function setUp(): void
         $this->configParams[self::CAMPAIGN_SUMMARY_PARAM] = in_array($this->getName(), $functionForUseSummary);
         $this->configParams[self::CAMPAIGN_RANGE_PARAM]   = in_array($this->getName(), $functionForUseRange);
         parent::setUp();
-        $this->campaignModel                                           = static::getContainer()->get('mautic.model.factory')->getModel('campaign');
+
+        $model = static::getContainer()->get('mautic.model.factory')->getModel('campaign');
+        \assert($model instanceof CampaignModel);
```
Comment (escopecz) [inline app/bundles/CampaignBundle/Tests/Controller/CampaignControllerFunctionalTest.php]: Not necessary to fix:

I'd like to get rid of the ModelFactory some time in the future. We should get the models from the container directly. There is no reason to do it from the ModelFactory. I know it was done like that before but since we are touching the code...
Ideal scenario:
```php
$model = static::getContainer()->get(CampaignModel::class);
```
If that doesn't work:
```php
$model = static::getContainer()->get(''mautic.campaign.model.campaign');
```

File: app/bundles/CoreBundle/Factory/MauticFactory.php [hunk 27]
```diff
@@ -324,15 +336,15 @@ public function getVersion()
...
     public function getLogger($system = false)
     {
         if ($system) {
-            return $this->container->get('logger');
-        } else {
-            return $this->container->get('monolog.logger.mautic');
+            return $this->logger;
         }
+
+        return $this->container->get('monolog.logger.mautic');
```
Comment (escopecz) [inline app/bundles/CoreBundle/Factory/MauticFactory.php:347]: I don't require this change as this class is deprecated, but:

I'd just return `$this->logger`. I see no reason for the if above. WDYT?

File: app/bundles/PageBundle/Tests/Controller/PublicControllerTest.php [hunk 28]
```diff
@@ -266,8 +265,10 @@ private function getVariantContent($aCount, $bCount, $cCount)
             );
 
         $container = $this->createMock(Container::class);
-        $container->method('has')
-            ->will($this->returnValue(true));
+        $container->expects(self::never())
+            ->method('has'); // THIS CAN FAIL!!!!!!!!!!!!!!
```
Comment (escopecz) [inline app/bundles/PageBundle/Tests/Controller/PublicControllerTest.php]: What shall we do with this? Do you suggest to merge this or is this just a type of "todo" comment?

File: app/bundles/CoreBundle/Factory/MauticFactory.php (hunk 27 above)
Comment (biozshock) [reply app/bundles/CoreBundle/Factory/MauticFactory.php:347]: The logger in the `$this->logger` is the "main" logger in the config, while "mautic" logger has different setting, meaning those objects are not equal and can log into different sources.

File: app/bundles/PageBundle/Tests/Controller/PublicControllerTest.php (hunk 28 above)
Comment (biozshock) [reply app/bundles/PageBundle/Tests/Controller/PublicControllerTest.php]: Yes, sorry, that was a comment for me. Removed.

File: app/bundles/CampaignBundle/Tests/Controller/CampaignControllerFunctionalTest.php (hunk 26 above)
Comment (biozshock) [reply app/bundles/CampaignBundle/Tests/Controller/CampaignControllerFunctionalTest.php]: Changed to the first variant.

**Review by escopecz — DISMISSED** [review]
Thank you! This is really great work. It reduces the tech debt in the baseline file by ~10%. 🎉

**Comment by escopecz** [comment]
@biozshock could you please rebase on 6.x branch and let's get merge it there.

**Comment by biozshock** [comment]
@escopecz done. You probably need to merge the PR, as this is heading to a year long one :)

**Review by escopecz — APPROVED** [review]
Thank you!
